File: 06_Dynamic_Game_Ch10_11/AM2007_R/code/functions/CCP2LogTransform.R
CCP2LogTransform <- function(x) {
  # input: 8 x 1 CCP essence vector
  # output: 8 x 3 logged CCP matrix

  output <- matrix(
    c(
      0, log(x[1]), log(1 - x[1]),
      log(1 - x[2]), log(x[2]), 0,
      0, log(x[3]), log(1 - x[3]),
      log(1 - x[4]), log(x[4]), 0,
      0, log(x[5]), log(1 - x[5]),
      log(1 - x[6]), log(x[6]), 0,
      0, log(x[7]), log(1 - x[7]),
      log(1 - x[8]), log(x[8]), 0
    ),
    ncol = 3, byrow = TRUE
  )
  output
}

File: 06_Dynamic_Game_Ch10_11/code_from_matlab/CCP2LogTransform.m
function output = CCP2LogTransform(x) 

% x is an 8*1 vector in the example and we make it an 8*3 matrix 
output =    [   0           log(x(1))    log(1-x(1)); 
                log(1-x(2)) log(x(2))    0     ; 
                0           log(x(3))    log(1-x(3)); 
                log(1-x(4)) log(x(4))    0     ;
                0           log(x(5))    log(1-x(5)); 
                log(1-x(6)) log(x(6))    0     ; 
                0           log(x(7))    log(1-x(7)); 
                log(1-x(8)) log(x(8))    0          ];
